backend/mora/auth/base.py

```diff
@@ -14,45 +14,23 @@
 progress.
 
 '''
-import base64
 import os
-import zlib
 
 import flask
-import requests.auth
+
+import flask_saml_sso
 
 __all__ = (
-    'SAMLAuth',
     'get_user',
-    'pack'
 )
 
-TOKEN_KEY = 'MO-Token'
-
 basedir = os.path.dirname(__file__)
 
 blueprint = flask.Blueprint('authentication', __name__,
                             url_prefix='/service', root_path=basedir)
 
 
-class SAMLAuth(requests.auth.AuthBase):
-    def __init__(self, assertion=None):
-        self.assertion = assertion
-
-    def __call__(self, r):
-        if self.assertion:
-            assertion = self.assertion
-        elif flask.session:
-            assertion = flask.session.get(TOKEN_KEY)
-        else:
-            assertion = None
-
-        if assertion:
-            r.headers['Authorization'] = assertion
-
-        return r
-
-
+@flask_saml_sso.requires_auth
 @blueprint.route('/user', methods=['GET'])
 def get_user():
```

Comment:
Perhaps we should fold this into the module as well?

Reply:
It's a thing that's only specific for OS2MO, and additionally requires configuration of which field in the attributes contains the name, so it isn't a thing that would work out-of-the-box anyway.

Reply:
But what we _could_ do, is make it easier to get the attributes from the session, so you don't have to remember that attributes are saved in `samlUserdata`. I'll do something about that.